assets/js/util.js

```diff
@@ -293,13 +299,79 @@ function callApy(options, endpoint) {
     var requestUrl = window.location.protocol + window.location.hostname +
         window.location.pathname + '?' + $.param(requestOptions.data);
 
+    apyRequestStartTime = Date.now();
     if(requestUrl.length > THRESHOLD_REQUEST_URL_LENGTH) {
         requestOptions.type = 'POST';
         return $.ajax(requestOptions);
     }
     return $.jsonp(requestOptions);
 }
 
+function checkServiceLoadTimes(requestDuration) {
+    var individualDurationThreshold = 2000;
+    var cumulativeDurationThreshold = 1500;
+    var lastNRequestsDuration = [];
+    var storedLastNRequestsDuration = [];
+    var demoThreshold = 10000; // for demo purposes only
+    try {
+        window.sessionStorage;
+
+        var oldRequestsMade = Number(sessionStorage.requestsMade);
+        sessionStorage.requestsMade = oldRequestsMade ? oldRequestsMade + 1 : 1;
+
+        var oldCumulativeRequestsTime = Number(sessionStorage.cumulativeRequestsTime);
+        sessionStorage.cumulativeRequestsTime = oldCumulativeRequestsTime ? oldCumulativeRequestsTime + requestDuration : requestDuration;
+
+        if(sessionStorage.requestsMade >= lastNRequests) {
+            sessionStorage.requestsMade = lastNRequests;
+            storedLastNRequestsDuration = JSON.parse(sessionStorage.getItem('lastNRequestsDuration'));
+            sessionStorage.cumulativeRequestsTime = Number(sessionStorage.cumulativeRequestsTime) - storedLastNRequestsDuration[0]; // move the window
+            lastNRequestsDuration.shift();
+            lastNRequestsDuration.push(requestDuration);
+            sessionStorage.setItem('lastNRequestsDuration', JSON.stringify(lastNRequestsDuration));
+        }
+        else {
+            lastNRequestsDuration.push(requestDuration);
+            sessionStorage.setItem('lastNRequestsDuration', JSON.stringify(lastNRequestsDuration));            
+        }
+        
+        var averageRequestsDuration = (Number(sessionStorage.cumulativeRequestsTime) / Number(sessionStorage.requestsMade));
+
+        if(requestDuration > individualDurationThreshold || averageRequestsDuration > cumulativeDurationThreshold) {
+            displayUnobtrusiveWarning();
+        }
+        if(averageRequestsDuration < demoThreshold) { // for demonstration , will remove it later
+            displayUnobtrusiveWarning();
+        }
+    }
+    catch(e) {
```

Comment:
Can we move the catch to just be around the `window.sessionStorage` and then `return` inside the `catch`? Less nesting that way.